docs/orchestration.md

```diff
@@ -278,6 +277,61 @@ Even though **stack-c** defined that it needs to be run after **stack-b**, since
 **runtime** order.
 
 
+## Execution Environment
+
+The environment in which stacks are going to be executed can be configured
+using the `terramate.config.run.env` block. This block has no labels and
+accepts arbitrary attributes where each attribute represents an environment
+variable that will be exported when executing the stack.
+The attributes must always evaluate to strings.
+
+Example:
+
+```hcl
+terramate {
+  config {
+    run {
+      env {
+        TF_PLUGIN_CACHE_DIR = "/some/path/etc"
+      }
+    }
+  }
+}
+```
+
+Will export the environment variable `TF_PLUGIN_CACHE_DIR` on the stack
+execution environment with the value `/some/path/etc`.
+
+Inside the `terramate.config.run.env` block the `env` namespace will be
+available including environment variables exported from the host:
+
+```hcl
+terramate {
+  config {
+    run {
+      env {
+        TF_PLUGIN_CACHE_DIR = "${env.HOME}/.terraform-cache-dir"
+      }
+    }
+  }
+}
+```
+
+The `env` namespace is read-only and is only available when evaluating
```

Comment:
it's strange to say that the `env` namespace is read-only if this config is for updating it. Do you mean, the `env` namespace is read-only inside the `.run.env` config? if so, there's no way to update a variable from an expression (AFAIK).

or maybe you mean that you cannot create variables in the `env` and reference the newly created variable inside this block? Like the config below, is it supported by current spec?

```
terramate {
  config {
    run {
      env {
        A = "test"
        B = "other ${A}"
      }
    }
  }
}
```

Reply:
I meant that the attributes defined inside the env block do not affect the `env` namespace (they only affect the actual environment when running commands), the namespace only exports the current environment from the host (hence read only). I'll try to make this more clear, I had a more elaborate explanation before but I remove it because I thought it complicated things...apparently I was wrong :laughing:

Reply:
Tried to improve the docs around this + added another example